syntax = "proto3";

package proto.ext.clipboard;

message ClipboardHtmlContent {
  string html = 1;
  string text = 2;
};

message ClipboardPathContent {
  string path = 1;
};

message ClipboardOptions {
  bool concealed = 1;
};

message ClearRequest {
};

message ClearResponse {};

message ReadContentRequest {
};

message ReadContentResponse {
  ClipboardReadContent content = 1;
};

message CopyToClipboardRequest {
  ClipboardContent content = 1;
  ClipboardOptions options = 2;
};

message PasteToClipboardRequest {
  ClipboardContent content = 1;
};

message ClipboardContent {
  oneof content {
    string text = 1;
    ClipboardHtmlContent html = 2;
    ClipboardPathContent path = 3;
  };
};

message ClipboardReadContent {
  string text = 1;
  optional string file = 2;
  optional string html = 3;
};

message CopyToClipboardResponse {
};

message PasteToClipboardResponse {
};

message Request {
  oneof payload {
    CopyToClipboardRequest copy = 1;
    PasteToClipboardRequest paste = 2;
    ReadContentRequest read_content = 3;
    ClearRequest clear = 4;
  };
};

message Response {
  oneof payload {
    CopyToClipboardResponse copy = 1;
    PasteToClipboardResponse paste = 2;
    ReadContentResponse read_content = 3;
    ClearResponse clear = 4;
  };
};
